base/set.jl

```diff
@@ -195,6 +195,25 @@ julia> unique(x -> x^2, [1, -1, 3, -3, 4])
  3
  4
 ```
+This functionality can also be used to extract the *indices* of the unique elements
+in an array:
+```jldoctest
+julia> a = [3,4,5,3,3,3,4,1];
+
+julia> i = unique(i -> a[i], eachindex(a))
+4-element Vector{Int64}:
+ 1
+ 2
+ 3
+ 8
+
+julia> a[i]
```

Comment:
Here this is `== unique(a)`. Is this guaranteed? 

If so, then perhaps it should say so. If not, maybe it should warn that the order could be different.

Reply:
I think it's guaranteed, because `unique(f, itr)` always returns "in the order that the first of each set of equivalent elements originally appears".

Reply:
added an `a[i] == unique(a)` example/test